packages/plugin-session-replay/src/session-replay.ts

```diff
@@ -0,0 +1,285 @@
+import { AMPLITUDE_PREFIX, BaseTransport } from '@amplitude/analytics-core';
+import { BrowserConfig, EnrichmentPlugin, Event, PluginType, Status } from '@amplitude/analytics-types';
+import * as IDBKeyVal from 'idb-keyval';
+import { record } from 'rrweb';
+import { shouldSplitEventsList } from './helpers';
+import { MAX_RETRIES_EXCEEDED_MESSAGE, STORAGE_FAILURE, SUCCESS_MESSAGE, UNEXPECTED_ERROR_MESSAGE } from './messages';
+import { Events, IDBStore, SessionReplayContext } from './typings/session-replay';
+
+const SESSION_REPLAY_SERVER_URL = 'https://api-secure.amplitude.com/sessions/track';
+const STORAGE_PREFIX = `${AMPLITUDE_PREFIX}_replay_unsent`;
+const PAYLOAD_ESTIMATED_SIZE_IN_BYTES_WITHOUT_EVENTS = 200; // derived by JSON stringifying an example payload without events
+const MAX_EVENT_LIST_SIZE_IN_BYTES = 20 * 1000000 - PAYLOAD_ESTIMATED_SIZE_IN_BYTES_WITHOUT_EVENTS;
+
+export class SessionReplayPlugin implements EnrichmentPlugin {
+  name = '@amplitude/plugin-session-replay';
+  type = PluginType.ENRICHMENT as const;
+  // this.config is defined in setup() which will always be called first
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  config: BrowserConfig;
+  storageKey = '';
+  retryTimeout = 1000;
+  events: Events = [];
+  currentSequenceId = 0;
+  private scheduled: ReturnType<typeof setTimeout> | null = null;
+  queue: SessionReplayContext[] = [];
+  stopRecordingEvents: ReturnType<typeof record> | null = null;
+  maxPersistedEventsSize = MAX_EVENT_LIST_SIZE_IN_BYTES;
+
+  async setup(config: BrowserConfig) {
+    config.loggerProvider.log('Installing @amplitude/plugin-session-replay.');
+
+    this.config = config;
+    this.storageKey = `${STORAGE_PREFIX}_${this.config.apiKey.substring(0, 10)}`;
+    await this.emptyStoreAndReset();
+
+    this.recordEvents();
+  }
+
+  execute(event: Event) {
+    event.event_properties = {
+      ...event.event_properties,
+      session_replay_enabled: true,
+    };
```

Comment:
curious if modifying the event props for $identify and $groupidentify events makes a difference

Reply:
Makes a difference in what way?

Reply:
I'll add that to our test plan! I haven't seen any issues in testing it myself